document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java

```diff
@@ -24,14 +26,35 @@ public DocumentInfo getDocumentInfo() {
         LOG.info("Started getting document");
 
         String resource = "";
+        String resourceId = "";
 
         Transaction transaction = transactionService.getTransaction(resource);
         if (transaction == null) {
             LOG.error("transaction not found");
             return null;
         }
 
-        return new DocumentInfo();
+        return Optional.of(transaction)
+                .map(Transaction::getResources)
+                .map(resources -> resources.get(resourceId))
+                .map(Resource::getLinks)
+                .map(links -> links.get("resource"))
+                // when abridged has been migrated to use the company-accounts api, the code for the
+                // company accounts should work for abridged, resulting in this abridged specific code
+                // qualifying for removal
+                .filter(this::isAbridged)
+                .map(accountsLinks -> new DocumentInfo())
+                .orElse(null);
 
     }
+
+    /**
+     * determines whether it is an abridged link as /accounts/ only exists within the abridged
+     * implementation
+     * @param accountLink - account link
+     * @return true if abridged, false if not
+     */
+    private boolean isAbridged(String accountLink) {
+        return accountLink.contains("/accounts/");
```

Comment:
Should we be using a regex matcher here to more confidently ensure it is an abridged account? May not be needed, just reduces the chance of following the wrong 'flow' in the case of bad data?

Reply:
added. I could have put it into a variable but to be honest its only going to be used in one place so we shouldn't have to worry.